crates/kornia-pnp/src/ippe.rs

```diff
@@ -0,0 +1,170 @@
+        points_image: &[[f32; 2]; 4],
+        k: &[[f32; 3]; 3],
+        square_length: f32,
+    ) -> Result<IPPEResult, PnPError> {
+        // Normalize the input image points using K^{-1}.
+        let k_mat = Mat3A::from_cols(
+            Vec3A::new(k[0][0], k[1][0], k[2][0]),
+            Vec3A::new(k[0][1], k[1][1], k[2][1]),
+            Vec3A::new(k[0][2], k[1][2], k[2][2]),
+        );
+        let k_inv = k_mat.inverse();
+        let image_points_norm: [[f32; 2]; 4] = points_image.map(|uv| {
+            let hp = k_inv * Vec3::new(uv[0], uv[1], 1.0);
+            [hp.x / hp.z, hp.y / hp.z]
+        });
+
+        // Build canonical 2D square points (object plane) in the order required above.
+        let h = (square_length as f64) / 2.0;
+        let src: [[f64; 2]; 4] = [[-h, h], [h, h], [h, -h], [-h, -h]];
+
+        // Target points are normalized image coordinates.
+        let dst: [[f64; 2]; 4] = [
+            [image_points_norm[0][0] as f64, image_points_norm[0][1] as f64],
+            [image_points_norm[1][0] as f64, image_points_norm[1][1] as f64],
+            [image_points_norm[2][0] as f64, image_points_norm[2][1] as f64],
+            [image_points_norm[3][0] as f64, image_points_norm[3][1] as f64],
+        ];
+
+        // Estimate homography mapping object-plane points to normalized image points.
+        let mut hmat = [[0.0f64; 3]; 3];
+        kornia_3d::pose::homography_4pt2d(&src, &dst, &mut hmat)
+            .map_err(|e| PnPError::SvdFailed(e.to_string()))?;
+
+        // Decompose homography into pose (assuming K = I because points are normalized).
+        let (r_mat, t_vec) = decompose_h_normalized(&hmat);
+
+        // Convert to arrays for result types and compute Rodrigues vector.
+        let r32 = [
+            [r_mat.x_axis.x, r_mat.y_axis.x, r_mat.z_axis.x],
+            [r_mat.x_axis.y, r_mat.y_axis.y, r_mat.z_axis.y],
+            [r_mat.x_axis.z, r_mat.y_axis.z, r_mat.z_axis.z],
+        ];
+        let t32 = [t_vec.x, t_vec.y, t_vec.z];
+
+        let rvec_v = SO3::from_matrix(&r_mat).log();
+        let rvec = [rvec_v.x, rvec_v.y, rvec_v.z];
+
+        // Compute RMS reprojection error under normalized intrinsics (K=I).
+        let obj3d = square_object_points(square_length);
+        let rmse1 = rmse_normalized(&obj3d, image_points_norm, &r32, &t32)?;
+
+        // For now, return the same solution twice as a placeholder; a subsequent
+        // iteration will implement the Jacobian-based second solution specific to IPPE.
+        let best = PnPResult {
+            rotation: r32,
+            translation: t32,
+            rvec,
+            reproj_rmse: Some(rmse1),
+            num_iterations: None,
+            converged: Some(true),
+        };
+
+        let second = best.clone();
+        Ok((best, second))
+    }
+}
+
+/// Compute the homography-based pose decomposition assuming normalized image coordinates (K = I).
+fn decompose_h_normalized(h: &[[f64; 3]; 3]) -> (Mat3A, Vec3) {
```

Comment:
should return/support an isometry struct to have more semantics around rotationt/translation? @cjpurackal @sidd-27